app/src/store/user.ts

```diff
@@ -67,8 +67,8 @@ export const userSlice = createSlice({
 });
 
 const auth0 = new Auth0({
-  domain: config.domain,
-  clientId: config.clientId,
+  domain: Config.DOMAIN_NAME!,
```

Comment:
Should we check if these are defined or not, instead of using the post-fix `!` operator?

Reply:
These should required.

Reply:
<img width="650" alt="Screen Shot 2023-05-11 at 10 08 57 AM" src="https://github.com/twilio/twilio-voice-react-native-app/assets/35968892/6066a893-3d6c-4fc6-b157-177a3f225945">
Without the `!` I get a type error. 
The type is defined in the package:
<img width="475" alt="Screen Shot 2023-05-11 at 10 13 03 AM" src="https://github.com/twilio/twilio-voice-react-native-app/assets/35968892/3eed95e0-dd69-445d-a7bc-51a775f98077">
Any suggestions?

Reply:
Let's try a type-guard before we use it. We should probably make some `util/env` module that does this checking for us.
```ts
if ('domain' in Config && typeof Config.domain !== 'undefined') {
  ...
}
```
Should make the compiler happy.

Reply:
I'm all for checking things like this but for this situation, what is the benefit other than making the compiler happy?
We know that these values are required but adding checks like this assumes they can be optional.

Reply:
I think the usage of `!` operator in this case is appropriate. It tells the compiler that this value cannot be undefined or null and trust that it should be there. I believe this is the definition of this operator correct?

Reply:
@kpchoy , do we get an error if the environment variables are not set? Does it fail silently or do we get a hint that the values are missing? If it fails silently, I'm +1 on Michael's suggestion, but we need to throw an error letting the dev know that the value is missing.

Reply:
Yes we do get a warning if env variables are not set. If I dont include the CLIENT_ID, this is the error:

<img width="896" alt="Screen Shot 2023-05-11 at 12 03 38 PM" src="https://github.com/twilio/twilio-voice-react-native-app/assets/35968892/475620e6-8855-46a2-bf01-161509ea1068">

Reply:
Which module is throwing this error? Is it getting all the way to the Auth0 module first, and then once Auth0 recognizes that the env var wasn't passed, it throws? My thought process is that our env helper module would parse the environment for all the required variables and then throw there instead of relying on dependencies to throw, sort of how the Reference Server is doing it.

Reply:
> Which module is throwing this error? Is it getting all the way to the Auth0 module first, and then once Auth0 recognizes that the env var wasn't passed, it throws? My thought process is that our env helper module would parse the environment for all the required variables and then throw there instead of relying on dependencies to throw, sort of how the Reference Server is doing it.

Thanks for providing more details @mhuynh5757 . I like this better.

Reply:
I have implemented the helper idea 👍